et/cs-229-machine-learning-tips-and-tricks.md

```diff
@@ -0,0 +1,289 @@
+
+<br>
+
+**33. Regularization ― The regularization procedure aims at avoiding the model to overfit the data and thus deals with high variance issues. The following table sums up the different types of commonly used regularization techniques:**
+
+&#10230; Regulariseerimine - Regulariseerimisprotsessi eesmärk on vältida mudeli ülemäärast ülesobitust treeningandmestikule ning kõrge dispersiooni probleemidega. Järgmises tabelis on toodud levinumate reguleerimist meetodite tüübid:
+
+<br>
+
+**34. [Shrinks coefficients to 0, Good for variable selection, Makes coefficients smaller, Tradeoff between variable selection and small coefficients]**
+
+&#10230; [Kahandab koefitsiente 0-ni, Hea muutujate valimiseks, Muudab koefitsendid väiksemaks, Muutuja valiku ja väikeste koefitsentide kompromiss]
+
+<br>
+
+**35. Diagnostics**
+
+&#10230; Diagnostika
+
+<br>
+
+**36. Bias ― The bias of a model is the difference between the expected prediction and the correct model that we try to predict for given data points.**
+
+&#10230; Nihe (Vabaliige) - Mudeli vabaliige on erinevus eeldatava ennustuse ja õige mudeli vahel, mida proovime antud andmepunktidega ennustada.
+
+<br>
+
+**37. Variance ― The variance of a model is the variability of the model prediction for given data points.**
+
+&#10230; Variatsioon - Mudeli variatsioon on mudeli ennustuse ja antud andmepunkti varieeruvus.
+
+<br>
+
+**38. Bias/variance tradeoff ― The simpler the model, the higher the bias, and the more complex the model, the higher the variance.**
+
+&#10230; Nihke/Variatsioon kompromiss - Mida lihtsam mudel seda suurem on nihe (vabaliige) ning mida keerulisem on mudel seda suurem on dispersioon.
+
+<br>
+
+**39. [Symptoms, Regression illustration, classification illustration, deep learning illustration, possible remedies]**
+
+&#10230; [Sümptomid, Regressiooni näide, Klassifikatsiooni näide, Süvavõppe näide, võimalikud abinõud]
+
+<br>
+
+**40. [High training error, Training error close to test error, High bias, Training error slightly lower than test error, Very low training error, Training error much lower than test error, High variance]**
+
+&#10230; [Kõrge treenimisviga, Treenimisviga sarnane tesimisvegaga, Kõrge nihe, Treenimisviga madalam kui testimisviga, Väga madal treenimisviga, Treenimisviga palju madalam kui testimisviga, Kõrge variatsioon]
+
+<br>
+
+**41. [Complexify model, Add more features, Train longer, Perform regularization, Get more data]**
+
+&#10230; [Mudeli keerukamaks muutmine, Muutujate lisamine, Pigem treenimine, Regulatsiooni lisamine, Andmehulga suurendamine]
+
+<br>
+
+**42. Error analysis ― Error analysis is analyzing the root cause of the difference in performance between the current and the perfect models.**
+
+&#10230; Veaanalüüs - Veaanalüüs on olemasoleva ja perfektse mudeli toivimuse erinevuse algpõhjuse leidmine.
```

Comment:
Consider using term "jõudlus", "suutlikkus" for performance, instead of "toimivus".